rfcs/i18n.md

```diff
@@ -0,0 +1,154 @@
+- Feature Name: i18n
+- Start Date: 2017-08-21
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This feature aims to make Gutenberg support multilingual sites.
+
+# Motivation
+[motivation]: #motivation
+
+Some sites are in multiple languages and want to share templates/content while only translating
+bits of the UI of the actual markdown content.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Overview
+
+### Configuration
+`language_code` is renamed to `default_language` as it is used for i18n mainly and is clearer that way.
+
+A map is added to the Config struct to hold some i18n strings:
+
+```toml
+default_language = "en"
+
+[languages]
+
+[languages.en]
+title = "My blog"
+description = "Hello world"
+
+[languages.fr]
+title = "Mon blog"
+```
+
+### Url
+
+The default language base url will be equal to `config.base_url`.
+Other languages will be available at `{config.base_url}/{language}`
```

Comment:
I think it’s important to have the language in the URL when i18n is used.

One challenge is when an user access the root url, what would happen?
I would address this issue by:

1. Copying the output folder in the root and in `/en`
2. All the links in the page direct to the link path with the language
3. Include a head tag `<link rel="canonical" href="https://example.com/en/" >`
4. Optional redirect in Javascript for the files in the root folder, like:

```jinja
{% if config.language_redirect %}
<script>
  if (navigator.languages) {
      let paths = ['/en/', '/fr/', '/jp/']
      let path = match_language_path(navigator.languages, paths, '/en/');
      window.location.replace(path);
  }
</script>
{% endif %}
```

Reply:
I think that should be up to the users whether they want the lang code in the url for their default language

Reply:
I agree it should be a choice. Gutenberg should make it easy either way. Do you mind adding a setting to opt in this behavior? `redirect_to_default_language`?

Reply:
For reference, the `index.html` of [rust-lang.org](https://www.rust-lang.org/) https://github.com/rust-lang/rust-www/blob/master/_layouts/redirect.html

It is a empty page with just the JS and HTML redirect. The setting could be then `force_i18n`.